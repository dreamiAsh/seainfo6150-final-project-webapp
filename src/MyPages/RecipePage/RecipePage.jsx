import React from "react";
import styles from "./RecipePage.module.css";
import { Link, BrowserRouter } from "react-router-dom";

const RecipePage = (props) => {

    return (
        <div className={styles.box}>
            <p className={styles.title}>{props.title}</p>
            <div className={styles.authorBox}>
                <BrowserRouter forceRefresh={true}><Link className={styles.author} to={`/author/${props.author}/${props.authorInfo}`}>By <span className={styles.authorInside}>{props.author}</span></Link></BrowserRouter>
                <p className={styles.reviews}>{props.reviews} Reviews</p>
                <p className={styles.likes}>{props.likes} Likes</p>
            </div>
            <div className={styles.imgBox}>
                <img className={styles.img} src={`https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%${props.imageId}.jpg&w=272&h=272&c=sc&poi=face&q=85`} alt="recipe image"/>
                <div className={styles.rightBox}>
                    <div className={styles.ingredientBox}>
                        <p className={styles.ingredientTitle}>Ingredient</p>
                        <p className={styles.ingredientContent}>{props.ingredient}</p>
                    </div>
                    <div className={styles.shortTextBox}>
                        <p className={styles.sign}>â€œ</p>
                        <p className={styles.shortText}>{props.shortText}</p>
                    </div>
                </div>
            </div>
            <div className={styles.content}>
                <p className={styles.step}>Step 1</p>
                <p>{props.step1}</p>
                <p className={styles.step}>Step 2</p>
                <p>{props.step2}</p>
            </div>
        </div>
    );
};

export default RecipePage;
