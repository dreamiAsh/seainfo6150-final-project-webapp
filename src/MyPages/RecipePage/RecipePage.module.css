.title{
    font-size: 28px;
    font-weight: bold;
    margin: 35px 60px 18px;
}
.authorBox{
    margin-left: 60px;
    margin-bottom: 40px;
    width: 90%;
    height: 25px;
}
.author{
    float: left;
    margin-right: 10%;
    color: gray;
    text-decoration: none;
}
.authorInside{
    color: black;
    border-bottom: solid 3px orange;
}
.authorInside:hover{
    color: grey;
}
.reviews{
    float: left;
    margin-right: 10%;
}
.likes{
    float: left;
}
.imgBox{
    margin-left: 60px;
    width: 94%;
    float: left;
}
.img{
    width: 40%;
    float: left;
}
.rightBox{
    float: left;
    width: 40%;
    margin-left: 3%;
}
.ingredientBox{
    width: 100%;
    height: 100px;
    border: solid 1px grey;
    padding: 20px;
    overflow: hidden;
}
.ingredientTitle{
    font-weight: bold;
    margin-bottom: 20px;
    font-size: 20px;
}
.shortTextBox{
    margin-top: 6%;
}
.sign{
    color: orange;
    font-size: 50px;
    height: 50px;
    line-height: 50px;
}
.shortText{
    margin-left: 40px;
}
.content{
    margin-left: 10%;
    margin-bottom: 100px;
    margin-top: 45px;
    width: 64%;
    float: left;
}
.step{
    font-size: 24px;
    font-weight: bold;
    width: 50%;
    margin-top: 35px;
    margin-bottom: 10px;
}
