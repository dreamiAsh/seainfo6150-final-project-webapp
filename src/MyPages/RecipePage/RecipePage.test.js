import React from "react";
import { render, fireEvent } from "@testing-library/react";
import RecipePage from "./RecipePage.jsx";

it("renders the RecipePage component correctly", () => {
  const { container } = render(<RecipePage />);
  expect(container).toMatchSnapshot();
});