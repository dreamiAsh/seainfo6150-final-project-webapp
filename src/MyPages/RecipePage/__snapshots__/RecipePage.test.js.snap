// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the RecipePage component correctly 1`] = `
<div>
  <div
    class="box"
  >
    <p
      class="title"
    />
    <div
      class="authorBox"
    >
      <a
        class="author"
        href="/author/undefined/undefined"
      >
        By 
        <span
          class="authorInside"
        />
      </a>
      <p
        class="reviews"
      >
         Reviews
      </p>
      <p
        class="likes"
      >
         Likes
      </p>
    </div>
    <div
      class="imgBox"
    >
      <img
        alt="recipe image"
        class="img"
        src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%undefined.jpg&w=272&h=272&c=sc&poi=face&q=85"
      />
      <div
        class="rightBox"
      >
        <div
          class="ingredientBox"
        >
          <p
            class="ingredientTitle"
          >
            Ingredient
          </p>
          <p
            class="ingredientContent"
          />
        </div>
        <div
          class="shortTextBox"
        >
          <p
            class="sign"
          >
            â€œ
          </p>
          <p
            class="shortText"
          />
        </div>
      </div>
    </div>
    <div
      class="content"
    >
      <p
        class="step"
      >
        Step 1
      </p>
      <p />
      <p
        class="step"
      >
        Step 2
      </p>
      <p />
    </div>
  </div>
</div>
`;
